# Практическая работа №2: Обработка выборочных данных. Нахождение точечных оценок параметров распределения

Выполнили студентки гр. 2383 Анищенко Анастасия и Бараева Елизавета. Вариант №24

## Цель работы

Получение практических навыков нахождения точечных статистических оценок параметров распределения.

## Основные теоретические положения

Начальный эмпирический момент k-го порядка:

$\bar{M}_k = \frac{1}{N}\sum n_j x_j^k$

Центральный эмпирический момент k-го порядка:

$\bar{m}_k = \frac{1}{N}\sum n_j (x_j - \bar{x}_в)^k,$
где $\bar{x}_в$ = $\bar{M}_1$.

Выборочная дисперсия:

$D_в = \bar{m}_2 = \frac{1}{N}\sum n_j(x_j - \bar{x}_в)^2$

Исправленная дисперсия:

$s^2 = \frac{N}{N-1}D_в$

Выборочное СКО:

$\sigma_в = \sqrt{D_в}$

Исправленное СКО:

$s = \sqrt{s^2}$

Статистическая оценка асимметрии:

$\bar{A}_s = \frac{\bar{m}_3}{s^3}$

Статистическая оценка эксцесса:

$\bar{E} = \frac{\bar{m}_4}{s^4} - 3$

Условная варианта:

$u_j = \frac{x_j - C}{h},$
где $C$ − условный ноль, значение которого выбирается равным значению варианты интервального ряда, являющейся средней или близкой к средней по значению в этом ряду.

Условный момент k-го порядка:

$\bar{M}_k^* = \frac{1}{N}\sum n_j\left( \frac{x_j-C}{h} \right)^k = \frac{1}{N}\sum n_j u_j^k$

Модой $M_o^*$ вариационного ряда называется варианта, которой соответствует наибольшая частота.

Формула для интервального ряда:

$M_o^* = x_{i-1} + h \cdot \frac{\tilde{m}_i - \tilde{m}_{i-1}}{(\tilde{m}_i - \tilde{m}_{i-1})+(\tilde{m}_i - \tilde{m}_{i+1})},$
где $h$ − длина интервалов; $\tilde{m}_i$ − относительная частота i-го интервала.

Медианой $M_e^*$ называется варианта, соответсвующая середине ранжированного ряда.

Формула для интервального ряда:

$M_e^* = x_{i-1} + h \cdot \frac{0.5 - \sum_{j=1}^{i-1} \tilde{m}_j}{\tilde{m}_i}$

Выборочный коэффициент вариации $V^*$ характеризует разброс значений вариант относительно выборочного среднего:

$V^* = \frac{\sigma_в}{\bar{x}_в} \cdot 100\%$

## Постановка задачи

Для заданных выборочных данных вычислить с использованием метода моментов и условных вариант точечные статистические оценки математического ожидания, дисперсии, среднеквадратичного отклонения, асимметрии, эксцесса, моды, медианы и коэффициента вариации исследуемой случайной величины. Полученные результаты содержательно проинтерпретировать.

## Выполнение работы

## Задание 1

In [83]:
import numpy as np
import pandas as pd
from tabulate import tabulate

# Интервальный ряд
interval_series = [
    {"i": 1, "left": 10.0, "right": 15.2, "x_i": 12.60, "m_i": 16, "rel_m_i": 0.1455, "acc_m_i": 16, "acc_rel_m_i": 0.1455},
    {"i": 2, "left": 15.2, "right": 20.5, "x_i": 17.85, "m_i": 22, "rel_m_i": 0.2000, "acc_m_i": 38, "acc_rel_m_i": 0.3455},
    {"i": 3, "left": 20.5, "right": 25.7, "x_i": 23.10, "m_i": 25, "rel_m_i": 0.2273, "acc_m_i": 63, "acc_rel_m_i": 0.5727},
    {"i": 4, "left": 25.7, "right": 30.9, "x_i": 28.30, "m_i": 23, "rel_m_i": 0.2091, "acc_m_i": 86, "acc_rel_m_i": 0.7818},
    {"i": 5, "left": 30.9, "right": 36.1, "x_i": 33.50, "m_i": 17, "rel_m_i": 0.1545, "acc_m_i": 103, "acc_rel_m_i": 0.9364},
    {"i": 6, "left": 36.1, "right": 41.4, "x_i": 38.75, "m_i": 4, "rel_m_i": 0.0364, "acc_m_i": 107, "acc_rel_m_i": 0.9727},
    {"i": 7, "left": 41.4, "right": 46.6, "x_i": 44.00, "m_i": 3, "rel_m_i": 0.0273, "acc_m_i": 110, "acc_rel_m_i": 1.0000}]

N = 110
h = 5.2
k = 7

df = pd.DataFrame(interval_series)
tabulate(df, headers='keys', tablefmt='grid', showindex=False)

# Вычисление средней арифметической частоты для выбора условного нуля
x_mean = np.sum(df['x_i'] * df['m_i']) / N
print(f"Средняя арифметическая частота: {x_mean:.2f}")

# Выбор условного нуля
idx_closest = np.argmin(np.abs(df['x_i'] - x_mean))
C = df.loc[idx_closest, 'x_i']
print(f"Условный ноль: C = {C}")

# Вычисление условных вариант
df['u_i'] = ((df['x_i'] - C) / h).round().astype(int)
df['n_i*u_i'] = df['m_i'] * df['u_i']
df['n_i*u_i^2'] = df['m_i'] * (df['u_i'] ** 2)
df['n_i*u_i^3'] = df['m_i'] * (df['u_i'] ** 3)
df['n_i*u_i^4'] = df['m_i'] * (df['u_i'] ** 4)
df['n_i*(u_i+1)^4'] = df['m_i'] * ((df['u_i'] + 1) ** 4)

tabulate(df[['i', 'x_i', 'm_i', 'u_i']], headers=['i', 'x_i', 'n_i', 'u_i'], tablefmt='grid', showindex=False)

# Создание строки с итоговыми суммами
sums = pd.DataFrame({
    'i': ['Σ'],
    'left': ['-'],
    'right': ['-'],
    'x_i': ['-'],
    'm_i': [df['m_i'].sum()],
    'u_i': ['-'],
    'n_i*u_i': [df['n_i*u_i'].sum()],
    'n_i*u_i^2': [df['n_i*u_i^2'].sum()],
    'n_i*u_i^3': [df['n_i*u_i^3'].sum()],
    'n_i*u_i^4': [df['n_i*u_i^4'].sum()],
    'n_i*(u_i+1)^4': [df['n_i*(u_i+1)^4'].sum()]})

df_with_sums = pd.concat([df, sums], ignore_index=True)

# Создание таблицы №1
print("\nТаблица №1:")
display_cols = ['i', 'x_i', 'm_i', 'u_i', 'n_i*u_i', 'n_i*u_i^2', 'n_i*u_i^3', 'n_i*u_i^4', 'n_i*(u_i+1)^4']
print(tabulate(df_with_sums[display_cols].rename(columns={'m_i': 'n_i'}), headers='keys', tablefmt='grid', showindex=False))

# Проведение контроля вычислений
sum_n = sums.loc[0, 'm_i']
sum_n_u = sums.loc[0, 'n_i*u_i']
sum_n_u2 = sums.loc[0, 'n_i*u_i^2']
sum_n_u3 = sums.loc[0, 'n_i*u_i^3']
sum_n_u4 = sums.loc[0, 'n_i*u_i^4']
sum_n_u_plus1_4 = sums.loc[0, 'n_i*(u_i+1)^4']

print("\nПроверка вычислений:")
control = sum_n + 4*sum_n_u + 6*sum_n_u2 + 4*sum_n_u3 + sum_n_u4
print(f"По формуле N + 4*Σnu + 6*Σnu² + 4*Σnu³ + Σnu⁴ = {sum_n} + 4*{sum_n_u} + 6*{sum_n_u2} + 4*{sum_n_u3} + {sum_n_u4} = {control}")

Средняя арифметическая частота: 24.36
Условный ноль: C = 23.1

Таблица №1:
+-----+-------+-------+-------+-----------+-------------+-------------+-------------+-----------------+
| i   | x_i   |   n_i | u_i   |   n_i*u_i |   n_i*u_i^2 |   n_i*u_i^3 |   n_i*u_i^4 |   n_i*(u_i+1)^4 |
+=====+=======+=======+=======+===========+=============+=============+=============+=================+
| 1   | 12.6  |    16 | -2    |       -32 |          64 |        -128 |         256 |              16 |
+-----+-------+-------+-------+-----------+-------------+-------------+-------------+-----------------+
| 2   | 17.85 |    22 | -1    |       -22 |          22 |         -22 |          22 |               0 |
+-----+-------+-------+-------+-----------+-------------+-------------+-------------+-----------------+
| 3   | 23.1  |    25 | 0     |         0 |           0 |           0 |           0 |              25 |
+-----+-------+-------+-------+-----------+-------------+-------------+-------------+--------

Контрольная сумма, полученная методом произведений, равна результату по формуле: $\sum n_j (u_j + 1)^4 = \sum n_j u_j^4 + 4 \sum n_j u_j^3 + 6 \sum n_j u_j^2 + 4 \sum n_j u_j + 1$

## Задание 2

Вывод формулы для центрального эмпирического мамента 1-го порядка:
$\bar m_1 = \frac{1}{N} \sum n_j (x_j - \bar x_в) = \frac{1}{N} \sum n_j x_j - \frac{1}{N} \sum n_j \bar x_в = \bar x_в - \bar x_в \cdot \frac{1}{N} \sum n_j = \bar x_в - \bar x_в \cdot 1 = 0$

In [84]:
# Вычисление условных эмпирических моментов
M1 = sum_n_u / N
M2 = sum_n_u2 / N
M3 = sum_n_u3 / N
M4 = sum_n_u4 / N

# Вычисление через условные моменты центральных эмпирических моментов
m1_central = 0
m2_central = (M2 - M1**2) * (h**2)
m3_central = (M3 - 3 * M1 * M2 + 2 * (M1**3)) * (h**3)
m4_central = (M4 - 4 * M1 * M3 + 6 * (M1**2) * M2 - 3 * (M1**4)) * (h**4)

# Создание таблицы №2
table2_data = {
    'k': [1, 2, 3, 4],
    '~M_k*': np.round([M1, M2, M3, M4], 2),
    '~m_k': np.round([m1_central, m2_central, m3_central, m4_central], 2)
}
table2 = pd.DataFrame(table2_data)

print("Таблица №2:")
print(tabulate(table2, headers=['k', '~M_k*', '~m_k'], tablefmt='grid', showindex=False, numalign='right'))

Таблица №2:
+-----+---------+---------+
|   k |   ~M_k* |    ~m_k |
+=====+=========+=========+
|   1 |    0.25 |       0 |
+-----+---------+---------+
|   2 |    2.37 |   62.53 |
+-----+---------+---------+
|   3 |    2.81 |  153.47 |
+-----+---------+---------+
|   4 |   15.14 | 9669.74 |
+-----+---------+---------+


## Задание 3

In [85]:
# Вычисление выборочных среднего и дисперсии с помощью стандартной формулы
x_v_standard = np.sum(df['x_i'] * df['m_i']) / N
D_v_standard = np.sum(df['m_i'] * (df['x_i'] - x_v_standard)**2) / N

print(f"Выборочное среднее: ~x_в = {x_v_standard:.2f}")
print(f"Выборочная дисперсия: D_в = {D_v_standard:.2f}")

Выборочное среднее: ~x_в = 24.36
Выборочная дисперсия: D_в = 63.16


Вывод формул выборочных среднего и дисперсии через условные варианты:

$\bar x_в = \bar M_1^* h + C = h \cdot \frac{1}{N} \sum n_j u_j + C$

$ D_в = \frac{1}{N} \sum n_j (x_j - \bar x_в)^2 = \frac{1}{N} \sum n_j \left( (C + h u_j) - \left( C + h \cdot \frac{1}{N} \sum n_j u_j \right) \right)^2 = \frac{1}{N} \sum n_j \left( h u_j - h \cdot \frac{1}{N} \sum n_j u_j \right)^2 = h^2 \cdot \frac{1}{N} \sum n_j \left( u_j - \frac{1}{N} \sum n_j u_j \right)^2 = h^2 \cdot \frac{1}{N} \sum n_j \left( u_j^2 - 2u_j \frac{1}{N} \sum n_j u_j + \left( \frac{1}{N} \sum n_j u_j \right)^2 \right) = h^2 \left( \frac{1}{N} \sum n_j u_j^2 - 2 \left( \frac{1}{N} \sum n_j u_j \right)^2 + \left( \frac{1}{N} \sum n_j u_j \right)^2 \right) = h^2 \left( \frac{1}{N} \sum n_j u_j^2 - \left( \frac{1}{N} \sum n_j u_j \right)^2 \right)$

In [86]:
# Вычисление выборочных среднего и дисперсии с помощью условных вариант
x_v_conditional = h * sum_n_u / N + C
D_v_conditional = (h**2) * ((sum_n_u2 / N) - (sum_n_u / N)**2)

print(f"Выборочное среднее: ~x_в = {x_v_conditional:.2f}")
print(f"Выборочная дисперсия: D_в = {D_v_conditional:.2f}")

Выборочное среднее: ~x_в = 24.38
Выборочная дисперсия: D_в = 62.53


In [87]:
# Вычисление выборочного среднего квадратического отклонения
sigma_v = np.sqrt(D_v_standard)
print(f"Выборочное СКО: σ_в = {sigma_v:.4f}")

Выборочное СКО: σ_в = 7.9473


Значения выборочных среднего и дисперсии, вычисленные с помощью стандартных формул и условных вариант, практически равны. Небольшие различия связаны с ошибками округления при записи промежуточных результатов.

## Задание 4

In [88]:
# Вычисление исправленной выборочной дисперсии
s2 = (N / (N - 1)) * D_v_standard
print(f"Исправленная выборочная дисперсия: s² = {s2:.2f}")

Исправленная выборочная дисперсия: s² = 63.74


In [89]:
# Вычисление исправленного СКО
s = np.sqrt(s2)
print(f"Исправленное СКО: s = {s:.4f}")

Исправленное СКО: s = 7.9836


Исправленная выборочная дисперсия получилась немного больше смещенной. Это связано с тем, что при вычислении смещенной дисперсии деление выполняется на $N$, следовательно, разброс данных немного уменьшается. Деление на $N-1$ позволяет компенсировать возникающее занижение оценки, поэтому исправленная дисперсия получается немного больше и точнее отражает разброс данных в генеральной совокупности.

Аналогичная ситуация наблюдается и для стандартных отклонений: смещенное СКО немного меньше исправленного. Однако различие между ними очень мало, так как отклонение вычисляется на основе дисперсии и корень смягчает эту разницу.

## Задание 5

In [90]:
# Нахождение статистической оценки коэффициента асимметрии
As = m3_central / (s**3)
print(f"Коэффициент асимметрии: As = {As:.2f}")

Коэффициент асимметрии: As = 0.30


In [91]:
# Нахождение статистической оценки коэффициента эксцесса
E = m4_central / (s**4) - 3
print(f"Коэффициент эксцесса: E = {E:.2f}")

Коэффициент эксцесса: E = -0.62


Положительное значение коэффициента асимметрии указывает на правую асимметрию распределения, то есть распределение слегка вытянуто вправо, а основная масса данных смещена немного левее центра. При этом форма распределения близка к симметричной, так как значение коэффициента близко к 0.

Отрицательное значение коэффициента эксцесса указывает на плосковершинность распределения по сравнению с нормальным. Его вершина менее выражена, а данные более равномерно распределены вокруг среднего, без ярко выраженного пика.

## Задание 6

In [92]:
# Вычисление моды
modal_idx = df['m_i'].idxmax()
modal_interval = df.loc[modal_idx]

x_low = modal_interval['left']  # нижняя граница модального интервала
f_mod_int = modal_interval['rel_m_i']  # частота модального интервала
f_prev_mod_int = df.loc[modal_idx - 1, 'rel_m_i'] if modal_idx > 0 else 0  # частота интервала перед модальным
f_next_mod_int = df.loc[modal_idx + 1, 'rel_m_i'] if modal_idx < k - 1 else 0  # частота интервала после модального

Mo = x_low + h * (f_mod_int - f_prev_mod_int) / ((f_mod_int - f_prev_mod_int) + (f_mod_int - f_next_mod_int))
print(f"Мода: Mo = {Mo:.2f}")

Мода: Mo = 23.62


In [93]:
# Вычисление медианы
median_idx = df[df['acc_m_i'] >= N / 2].index[0]
median_interval = df.loc[median_idx]

x_low_med = median_interval['left']  # нижняя граница медианного интервала
f_med_int = median_interval['rel_m_i']  # частота медианного интервала
acc_prev_med_int = df.loc[median_idx - 1, 'acc_rel_m_i'] if median_idx > 0 else 0  # накопленная частота перед медианным интервалом

Me = x_low_med + h * (0.5 - acc_prev_med_int) / f_med_int
print(f"Медиана: Me = {Me:.2f}")

Медиана: Me = 24.03


In [94]:
# Вычисление коэффициента вариации
V = (sigma_v / x_mean) * 100
print(f"Коэффициент вариации: V* = {V:.2f}")

Коэффициент вариации: V* = 32.63


Значение моды меньше значения медианы (23,62 < 24,03) это говорит о том, что распределение имеет умеренный правосторонний уклон: наиболее часто встречающееся значение расположено чуть левее центра распределения.

Коэффициент вариации характеризует относительную изменчивость признака. Его полученное значение указывает на значительную степень рассеивания данных. Значение коэффициента вариации не превышает 33%, следовательно, совокупность является однородной.

## Выводы

В ходе выполнения лабораторной работы на основе выборочных данных были вычислены точечные статистические оценки параметров распределения. Были введены условные варианты, на основе которых рассчитаны условные эмпирические моменты. С использованием соотношений между условными и центральными моментами определены центральные эмпирические моменты. Корректность проведенных расчетов подтверждена практически полным совпадением значений выборочных среднего и дисперсии, найденных двумя способами. По полученным данным было вычислено выборочное СКО, а также исправленные оценки дисперсии и СКО. Сравнение смещенных и несмещенных оценок показало, что исправленная дисперсия незначительно превышает выборочную. На основе центральных моментов третьего и четвертого порядков были рассчитаны коэффициенты асимметрии и эксцесса. Полученные значения позволили охарактеризовать форму распределения: выявлено наличие правосторонней асимметрии и плосковершинность по сравнению с нормальным распределением. Также были определены мода и медиана. Сопоставление этих параметров подтвердило наличие правосторонней асимметрии. Рассчитанный коэффициент вариации показал, что совокупность является однородной, а вариация признака — умеренной.
